common/logger/src/metadata.rs

```diff
@@ -23,24 +24,34 @@ pub struct Metadata {
 
     /// The file name and line number together 'file:line'
     location: &'static str,
+
+    /// The program backtrace taken when the event occurred. Backtraces are
+    /// only supported for errors.
+    backtrace: Option<Backtrace>,
```

Comment:
Luckily this makes it at the end of the log lines!  We might want to consider trimming it in some way.  Usually only the first few lines in a stack trace are useful.

Reply:
Yeah, I noted this in the PR description. If we decide to go with this approach, I can trim off the first few stack frames to make the backtrace clearer 😄

Reply:
while I generally agree I would avoid pre-optimizations, I don't think our error logs are that frequent and we could also look at sampling rather than trying to guess what portion of a log might be valuable.